src/v2/BaseLoginRouter.js

```diff
@@ -66,6 +66,7 @@ export default Router.extend({
       // and then the open tooltip will lose focus and close.
     });
 
+    this.hooks = options.hooks;
```

Comment:
since hooks are coming from `config` we should put it on `this.settings.hooks` instead..

Reply:
hooks *can* be defined in config, but they can also be set at runtime either before or after render. Since settings is created by the router (after render), it could not hold hook definitions added at runtime before render. For this reason I broke it apart into its own model.